src/ui/components/search/autocompletecomponent.js

```diff
@@ -192,10 +192,20 @@ export default class AutoCompleteComponent extends Component {
       sectionIndex: this._sectionIndex,
       resultIndex: this._resultIndex,
       promptHeader: this._originalQuery.length === 0 ? this.promptHeader : null,
-      listLabelIdName: this.listLabelIdName
+      listLabelIdName: this.listLabelIdName,
+      eventOptions: this.eventOptions(data)
     }));
   }
 
+  eventOptions (data) {
+    const eventOptions = data.sections?.map(section =>
+      section.results.map(result => {
+        return { suggestedSearchText: result.value };
```

Comment:
Are we sure that `results` is always an array (as opposed to undefined)? And that `section` and `result` are always objects? It looks like that's the case, but if it's not this could crash at runtime

Reply:
from the methods in `autocompleteresponsetransformer.js`, it looks like `section` and `result` should always be objects. The way we generate `results` field there does make sure it's always be an array. But I do see `hasResults()` in autocomplete check if it's undefined anway, so I will add an optional chain guard for results here.